localize.m

```diff
@@ -0,0 +1,90 @@
+    NSDictionary<NSString *, NSString *> *jsonDictionary = [NSJSONSerialization JSONObjectWithData:fileData options:0 error:nil];
+    NSMutableDictionary<NSString *, NSDictionary<NSString *, NSString *> *> *langJsonMap = [NSMutableDictionary dictionary];
+    NSMutableSet<NSString *> *keySet = [NSMutableSet set];
+    [jsonDictionary enumerateKeysAndObjectsUsingBlock:^(id key, id value, BOOL* stop) {
+        NSString *lang = key;
+        NSString *putilsLocalizableStringsFile = value;
+        NSDictionary<NSString *, NSString *> *localizationValuesMap = [LocalizationHelper createLocalizationValuesMapping:putilsLocalizableStringsFile];
+        [keySet addObjectsFromArray:localizationValuesMap.allKeys];
+        langJsonMap[lang] = localizationValuesMap;
+    }];
+    LocalizationResult *result = [[LocalizationResult alloc] init];
+    result.keySet = keySet;
+    result.languageCodeToLocalizationsMap = langJsonMap;
+    return result;
+}
+
++ (NSDictionary<NSString *, NSString *> *)createLocalizationValuesMapping:(NSString *)putilsLocalizableStringsFile {
+    return [NSJSONSerialization JSONObjectWithData:[NSData dataWithContentsOfFile:putilsLocalizableStringsFile options:0 error:nil] options:0 error:nil];
+}
+
++ (void)writeLanguageMaps:(LocalizationResult *)result sortedKeySetArray:(NSArray<NSString *> *)sortedKeySetArray valuesJsonLzfseOutputMap:(NSDictionary<NSString *, NSString *> *)valuesJsonLzfseOutputMap compressToolPath:(NSString *)compressToolPath {
+    [result.languageCodeToLocalizationsMap enumerateKeysAndObjectsUsingBlock:^(id key, id value, BOOL* stop) {
+        NSString *lang = key;
+        NSDictionary<NSString *, NSString *> *localizationValuesMap = value;
+        NSMutableArray<NSString *> *sortedValues = [NSMutableArray array];
+        for (NSString *key in sortedKeySetArray) {
+            NSString *value = localizationValuesMap[key];
+            if (value) {
+                [sortedValues addObject:value];
+            }
+        }
+        NSString *outputPath = valuesJsonLzfseOutputMap[lang];
+        [LocalizationHelper writeCompressedData:[NSJSONSerialization dataWithJSONObject:sortedValues options:0 error:nil] outputPath:outputPath compressToolPath:compressToolPath];
+    }];
+}
+
++ (void)writeCompressedData:(NSData *)data outputPath:(NSString *)outputPath compressToolPath:(NSString *)compressToolPath {
+    NSURL *outputPathURL = [NSURL fileURLWithPath: outputPath];
+    NSString *tempFilePath = [NSTemporaryDirectory() stringByAppendingPathComponent:outputPathURL.lastPathComponent];
+    [data writeToURL:[NSURL fileURLWithPath: tempFilePath] atomically:YES];
+    NSTask *task = [[NSTask alloc] init];
+    task.launchPath = compressToolPath;
+    task.arguments = @[tempFilePath, outputPath];
+    [task launch];
+    [task waitUntilExit];
+    [[NSFileManager defaultManager] removeItemAtPath:tempFilePath error:nil];
+}
+
+@end
+
+int main(int argc, char *argv[]) {
```

Comment:
Maybe this can use the existing localize.sh script?

Reply:
The bazel logic handles everything you have in that shell in a hermetic way and also sidesteps the need for ruby to run this (which is likely not needed since its essentially doing globbing)